### PR TITLE
57 admin user can view order

### PR DESCRIPTION
**Pull request by michael-reeves**
Add Admin::Order #show page.
Refactor Order #show page to pull out common elements into partial for use in both pages
Add preliminary partial for addresses
Add full_name method to User model

### CODE REVIEW

File: spec/features/admin_can_view_order_details_spec.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    order.order_items.create(product_id: plant2.id,
+                             quantity:   1,
+                             unit_price: plant2.price)
+
+
+    admin = User.create(first_name: "Mike",
+                        last_name:  "D",
+                        email:      "mike@mike.com",
+                        password:   "12345678",
+                        role:       "admin")
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                             unit_price: plant1.price)
+    order.order_items.create(product_id: plant2.id,
+                             quantity:   1,
+                             unit_price: plant2.price)
+
+
+    admin = User.create(first_name: "Mike",
+                        last_name:  "D",
+                        email:      "mike@mike.com",
+                        password:   "12345678",
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                             quantity:   3,
+                             unit_price: plant1.price)
+    order.order_items.create(product_id: plant2.id,
+                             quantity:   1,
+                             unit_price: plant2.price)
+
+
+    admin = User.create(first_name: "Mike",
+                        last_name:  "D",
+                        email:      "mike@mike.com",
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    order.order_items.create(product_id: plant1.id,
+                             quantity:   3,
+                             unit_price: plant1.price)
+    order.order_items.create(product_id: plant2.id,
+                             quantity:   1,
+                             unit_price: plant2.price)
+
+
+    admin = User.create(first_name: "Mike",
+                        last_name:  "D",
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    order = user.orders.create(status: "ordered",
+                       created_at: DateTime.civil(2015, 07, 05, 21, 33, 0),
+                       updated_at: DateTime.civil(2015, 07, 05, 21, 33, 0))
+
+    order.order_items.create(product_id: plant1.id,
+                             quantity:   3,
+                             unit_price: plant1.price)
+    order.order_items.create(product_id: plant2.id,
+                             quantity:   1,
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                       last_name:  "Doe",
+                       email:      "jane@doe.com",
+                       password:   "password")
+
+    order = user.orders.create(status: "ordered",
+                       created_at: DateTime.civil(2015, 07, 05, 21, 33, 0),
+                       updated_at: DateTime.civil(2015, 07, 05, 21, 33, 0))
+
+    order.order_items.create(product_id: plant1.id,
+                             quantity:   3,
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    plant2 = category.products.create(name: "Plant 2",
+                        description: "This is the description for plant 2",
+                        price:       29.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+
+    user = User.create(first_name: "Jane",
+                       last_name:  "Doe",
+                       email:      "jane@doe.com",
+                       password:   "password")
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                        status:      "active")
+    plant2 = category.products.create(name: "Plant 2",
+                        description: "This is the description for plant 2",
+                        price:       29.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+
+    user = User.create(first_name: "Jane",
+                       last_name:  "Doe",
+                       email:      "jane@doe.com",
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+    plant2 = category.products.create(name: "Plant 2",
+                        description: "This is the description for plant 2",
+                        price:       29.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+
+    user = User.create(first_name: "Jane",
+                       last_name:  "Doe",
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Spacing adds to readability

File: spec/features/admin_can_view_order_details_spec.rb [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+                       email:      "jane@doe.com",
+                       password:   "password")
+
+    order = user.orders.create(status: "ordered",
+                       created_at: DateTime.civil(2015, 07, 05, 21, 33, 0),
+                       updated_at: DateTime.civil(2015, 07, 05, 21, 33, 0))
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Line lengths prevent alignment

File: spec/features/admin_can_view_order_details_spec.rb [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+
+    user = User.create(first_name: "Jane",
+                       last_name:  "Doe",
+                       email:      "jane@doe.com",
+                       password:   "password")
+
+    order = user.orders.create(status: "ordered",
+                       created_at: DateTime.civil(2015, 07, 05, 21, 33, 0),
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Line lengths prevent alignment

File: spec/features/admin_can_view_order_details_spec.rb [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    plant1 = category.products.create(name: "Plant 1",
+                        description: "This is the description for plant 1",
+                        price:       19.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+    plant2 = category.products.create(name: "Plant 2",
+                        description: "This is the description for plant 2",
+                        price:       29.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Line lengths prevent alignment

File: spec/features/admin_can_view_order_details_spec.rb [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+    plant1 = category.products.create(name: "Plant 1",
+                        description: "This is the description for plant 1",
+                        price:       19.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+    plant2 = category.products.create(name: "Plant 2",
+                        description: "This is the description for plant 2",
+                        price:       29.99,
+                        image_url:   "plants/plant-2.jpg",
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Line lengths prevent alignment

File: spec/features/admin_can_view_order_details_spec.rb [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+    )
+
+    plant1 = category.products.create(name: "Plant 1",
+                        description: "This is the description for plant 1",
+                        price:       19.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+    plant2 = category.products.create(name: "Plant 2",
+                        description: "This is the description for plant 2",
+                        price:       29.99,
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Line lengths prevent alignment

File: spec/features/admin_can_view_order_details_spec.rb [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+      description: "Carnivorous plants!"
+    )
+
+    plant1 = category.products.create(name: "Plant 1",
+                        description: "This is the description for plant 1",
+                        price:       19.99,
+                        image_url:   "plants/plant-2.jpg",
+                        status:      "active")
+    plant2 = category.products.create(name: "Plant 2",
+                        description: "This is the description for plant 2",
```
Comment (michael-reeves) [inline spec/features/admin_can_view_order_details_spec.rb]: Line lengths prevent alignment

File: spec/features/admin_can_view_order_details_spec.rb (hunk 2 above)
Comment (imwithsam) [inline spec/features/admin_can_view_order_details_spec.rb]: Looks good. Appears lines 28-29 would spill over the 80 character limit if you moved it right.

File: spec/features/admin_can_view_order_details_spec.rb (hunk 3 above)
Comment (imwithsam) [inline spec/features/admin_can_view_order_details_spec.rb]: Looks good. Appears lines 28-29 would spill over the 80 character limit if you moved it right.

File: spec/features/admin_can_view_order_details_spec.rb (hunk 4 above)
Comment (imwithsam) [inline spec/features/admin_can_view_order_details_spec.rb]: Looks good. Appears lines 28-29 would spill over the 80 character limit if you moved it right.

File: spec/features/admin_can_view_order_details_spec.rb (hunk 5 above)
Comment (imwithsam) [inline spec/features/admin_can_view_order_details_spec.rb]: Looks good. Appears lines 28-29 would spill over the 80 character limit if you moved it right.

File: spec/features/admin_can_view_order_details_spec.rb (hunk 6 above)
Comment (imwithsam) [inline spec/features/admin_can_view_order_details_spec.rb]: Looks good. Appears lines 28-29 would spill over the 80 character limit if you moved it right.

File: spec/features/admin_can_view_order_details_spec.rb (hunk 7 above)
Comment (imwithsam) [inline spec/features/admin_can_view_order_details_spec.rb]: Looks good. Appears lines 12-15 and 17-20 would spill over the 80 character limit if you moved it right.

**Comment by imwithsam** [comment]
Looks good!

**Pull request merged**